agent/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java

```diff
@@ -0,0 +1,337 @@
+        if (!potentialSetters.isEmpty()) {
+          List<Method> pickedSetters =
+              data.pickValues(potentialSetters, data.consumeInt(0, potentialSetters.size()));
+          for (Method setter : pickedSetters) {
+            autofuzz(data, setter, obj);
+          }
+        }
+      }
+      return obj;
+    }
+    // We are out of more or less canonical ways to construct an instance of this class and have to
+    // resort to more heuristic approaches.
+
+    // First, try to find nested classes with names ending in Builder and call a subset of their
+    // chaining methods.
+    List<Class<?>> nestedBuilderClasses = getNestedBuilderClasses(type);
+    if (!nestedBuilderClasses.isEmpty()) {
+      Class<?> pickedBuilder = data.pickValue(nestedBuilderClasses);
+
+      List<Method> cascadingBuilderMethods = Arrays.stream(pickedBuilder.getMethods())
+                                                 .filter(m -> m.getReturnType() == pickedBuilder)
+                                                 .collect(Collectors.toList());
+
+      List<Method> originalObjectCreationMethods = getOriginalObjectCreationMethods(pickedBuilder);
+
+      int pickedMethodsNumber = data.consumeInt(0, cascadingBuilderMethods.size());
+      List<Method> pickedMethods = data.pickValues(cascadingBuilderMethods, pickedMethodsNumber);
+
+      Method builderMethod = data.pickValue(originalObjectCreationMethods);
+
+      Object builderObj = autofuzz(data, data.pickValue(pickedBuilder.getConstructors()));
+      for (Method method : pickedMethods) {
+        builderObj = autofuzz(data, method, builderObj);
+      }
+
+      try {
+        return builderMethod.invoke(builderObj);
+      } catch (Exception e) {
+        throw new AutofuzzConstructionException(e);
+      }
+    }
+
+    // We ran out of ways to construct an instance of the requested type. If in debug mode, report
+    // more detailed information.
+    if (!isDebug()) {
+      throw new AutofuzzConstructionException();
+    } else {
+      String summary = String.format(
+          "Failed to generate instance of %s:%nAccessible constructors: %s%nNested subclasses: %s%n",
+          type.getName(),
+          Arrays.stream(type.getConstructors())
+              .map(Utils::getReadableDescriptor)
+              .collect(Collectors.joining(", ")),
+          Arrays.stream(type.getClasses()).map(Class::getName).collect(Collectors.joining(", ")));
+      throw new AutofuzzConstructionException(summary);
+    }
+  }
+
+  static boolean isTest() {
+    String value = System.getenv("JAZZER_AUTOFUZZ_TESTING");
+    return value != null && !value.isEmpty();
+  }
+
+  static boolean isDebug() {
+    String value = System.getenv("JAZZER_AUTOFUZZ_DEBUG");
+    return value != null && !value.isEmpty();
+  }
+
+  private static String getDebugSummary(
+      Executable executable, Object thisObject, Object[] arguments) {
+    return String.format("%nMethod: %s::%s%s%nthis: %s%nArguments: %s",
```

Comment:
Should we add a prefix to the debug summary to make clear that it comes from Jazzer's autofuzz mode so that they become easier to spot?